nautobot/core/models/utils.py

```diff
@@ -16,10 +16,15 @@
 
 def array_to_string(array):
     """
-    Generate an efficient, human-friendly string from a set of integers. Intended for use with ArrayField.
+    Generate an efficient, human-friendly string from a set of integers or strings. Intended for use with ArrayField.
     For example:
         [0, 1, 2, 10, 14, 15, 16] => "0-2, 10, 14-16"
+        ["foo", "bar", "baz"] => "foo, bar, baz"
     """
+    if not array:
+        return ""
+    if isinstance(array[0], str):
+        return ", ".join(array)
```

Comment:
I don't think this is the right place for this change. Even though the name of this function suggests it's intended for converting arrays to strings it contains more specific logic for converting consecutive integers. Also, unless we have a use case for this to be used on arrays that could either be integers OR strings and we don't know ahead of time which one will be used, lets go ahead and just do the `", ".join(array)` on the fields where it's appropriate.